app/models/services/base-service.ts

```diff
@@ -168,17 +168,35 @@ export default abstract class BaseService<ServiceParamType> {
    *
    * @returns A promise resolving to the result of the callback.
    */
-  async invoke(
-    logger?: Logger, harmonyRoot?: string, requestUrl?: string,
+  async invokeOrAttach(
+    logger?: Logger, harmonyRoot?: string, requestUrl?: string, geojsonHash?: string,
   ): Promise<InvocationResult> {
     logger.info('Invoking service for operation', { operation: this.operation });
-    const job = this._createJob(requestUrl);
-    await this._createAndSaveWorkflow(logger, job);
+    const job = this._createJob(requestUrl, geojsonHash);
+    try {
+      await job.maybeAttach(db);
+      if (job.attachedStatus.didAttach) {
+        // attached jobs do not initiate new work
+        const { originalId, assumedId } = job.attachedStatus;
+        job.requestId = assumedId;
+        logger.info(`This job attached to a previous job: ${originalId} is now ${assumedId}.`);
+      } else {
+        // proceed with saving and creating workflow
+        await this._createAndSaveWorkflow(logger, job);
+      }
+    } catch (e) {
+      logger.error(e.stack);
+      throw new ServerError('Failed to save job to database.');
+    }
 
     const { isAsync, requestId } = job;
     this.operation.callback = `${env.callbackUrlRoot}/service/${requestId}`;
+    // If the current job attached to a running job then skip execution in the backend
+    this['maybeRun'] = job.attachedStatus.didAttach
```

Comment:
It looks like this might be the reason you changed the linter settings? I'm not entirely sure what it is going on here, but i don't think you need to set a variable on `this`.

Can't you just have:

```
const runFn = job.attachedStatus.didAttach...
 ...
runFn(logger).then...
 ```